src/pubsub.c

```diff
@@ -330,13 +332,13 @@ int pubsubUnsubscribeChannel(client *c, robj *channel, int notify, pubsubtype ty
 
 /* Unsubscribe all shard channels in a slot. */
 void pubsubShardUnsubscribeAllChannelsInSlot(unsigned int slot) {
-    if (!kvstoreDictSize(server.pubsubshard_channels, slot)) return;
+    if (!kvstoreHashtableSize(server.pubsubshard_channels, slot)) return;
 
-    kvstoreDictIterator *kvs_di = kvstoreGetDictSafeIterator(server.pubsubshard_channels, slot);
-    dictEntry *de;
-    while ((de = kvstoreDictIteratorNext(kvs_di)) != NULL) {
-        robj *channel = dictGetKey(de);
-        dict *clients = dictGetVal(de);
+    kvstoreHashtableIterator *kvs_di = kvstoreGetHashtableSafeIterator(server.pubsubshard_channels, slot);
+    void *element;
+    while (kvstoreHashtableIteratorNext(kvs_di, &element)) {
+        dict *clients = element;
+        robj *channel = *(robj **)clients->metadata;
```

Comment:
```suggestion
        robj *channel = (robj *)dictMetadata(clients);
```
There are a few places that can be cleaned up a bit.

Reply:
Thanks. It needs to be like this though:

```suggestion
        robj *channel = *(robj **)dictMetadata(clients);
```